CMakeLists.txt

```diff
@@ -324,10 +323,13 @@ endif()
 set_var_to_condition(MIOPEN_USE_COMGR_DEFAULT MIOPEN_EMBED_BUILD)
 option(MIOPEN_USE_COMGR "Use comgr to build kernels instead of offline tools" ${MIOPEN_USE_COMGR_DEFAULT})
 
+set_var_to_condition(MIOPEN_USE_MLIR_DEFAULT (NOT ${MIOPEN_EMBED_BUILD}))
+option(MIOPEN_USE_MLIR "Use MLIR compilation backend" ${MIOPEN_USE_MLIR_DEFAULT})
+
 if(MIOPEN_USE_MLIR)
     # REQUIRED is not supported before cmake 3.18
     find_library(LIBMLIRMIOPEN MLIRMIOpen REQUIRED)
-    if(NOT BUILD_SHARED_LIBS AND MIOPEN_USE_COMGR)
+    if(MIOPEN_EMBED_BUILD)
```

Comment:
Please revert this. Some of MIOpen users set BUILD_SHARED_LIBS=off and MIOPEN_USE_COMGR=on without explicit use of MIOPEN_EMBED_BUILD.

Reply:
I will revert this but just want to give you a heads up that it will block more scenarios than we actually should. When MIOpen is built as a static library but use COMGR as a dynamic library, things still work for MLIR backend. I believe `Smoke Aux 1::Fp32 Hip Static` stage is tracking this scenario.

Strictly speaking MLIR/COMGR combination will be an issue when and only when MIOpen is built as a static library to include all its dependencies in one binary. It looks to me that the client really should be use a specific cmake configuration (like `MIOPEN_EMBED_BUILD`) for this scenario.

So one day when MIOpen turn on COMGR by default, that Fp32 Hip Static stage will populate an error because it hit on this condition (whereas it can actually run through)

Reply:
Done in 6cad119

Reply:
Let's ask @JehandadKhan -- I am not sure which build configurations we are supporting right now etc.

Reply:
I agree with the current implementation, where we only block MLIR enablement when MIOpen is built fully statically. Embeded builds are ok if they link against MIOpen dynamically and therefore dynamically link against MLIR as well.